internal/controller/biossettings_controller.go

```diff
@@ -1143,30 +1149,42 @@ func (r *BiosSettingsReconciler) checkForRequiredPowerStatus(
 }
 
 func (r *BiosSettingsReconciler) checkIfMaintenanceGranted(
+	ctx context.Context,
 	log logr.Logger,
 	biosSettings *metalv1alpha1.BIOSSettings,
 	server *metalv1alpha1.Server,
-) bool {
+) (bool, metalv1alpha1.ServerMaintenanceState) {
```

Comment:
Could `bool` return value be removed and the call site simply checks for `state == InMaintenance`?